python/examples/workload/example.py

```diff
@@ -0,0 +1,181 @@
+        flop_count = self.M * self.N  # one addition per element
+        memory_reads = 2 * self.M * self.N * nbytes  # read A and B
+        memory_writes = self.M * self.N * nbytes  # write C
+        return (flop_count, memory_reads, memory_writes)
+
+    def payload_module(self):
+        with self.context, self.location:
+            float32_t = ir.F32Type.get()
+            shape = (self.M, self.N)
+            tensor_t = ir.RankedTensorType.get(shape, float32_t)
+            memref_t = ir.MemRefType.get(shape, float32_t)
+            mod = ir.Module.create()
+            with ir.InsertionPoint(mod.body):
+                args = [memref_t, memref_t, memref_t]
+                f = func.FuncOp(self.payload_function_name, (tuple(args), ()))
+                f.attributes["llvm.emit_c_interface"] = ir.UnitAttr.get()
+            with ir.InsertionPoint(f.add_entry_block()):
+                A = f.arguments[0]
+                B = f.arguments[1]
+                C = f.arguments[2]
+                a_tensor = bufferization.ToTensorOp(tensor_t, A, restrict=True)
+                b_tensor = bufferization.ToTensorOp(tensor_t, B, restrict=True)
+                c_tensor = bufferization.ToTensorOp(
+                    tensor_t, C, restrict=True, writable=True
+                )
+                add = linalg.add(a_tensor, b_tensor, outs=[c_tensor])
+                bufferization.MaterializeInDestinationOp(
+                    None, add, C, restrict=True, writable=True
+                )
+                func.ReturnOp(())
+        return mod
+
+    def schedule_module(self, dump_kernel=None, parameters=None):
+        with self.context, self.location:
+            schedule_module = ir.Module.create()
+            schedule_module.operation.attributes[
+                "transform.with_named_sequence"] = (ir.UnitAttr.get())
+            with ir.InsertionPoint(schedule_module.body):
+                named_sequence = transform.NamedSequenceOp(
+                    "__transform_main",
+                    [transform.AnyOpType.get()],
+                    [],
+                    arg_attrs=[{"transform.readonly": ir.UnitAttr.get()}],
+                )
+                with ir.InsertionPoint(named_sequence.body):
+                    anytype = transform.AnyOpType.get()
+                    func = match(named_sequence.bodyTarget, ops={"func.func"})
+                    mod = transform.get_parent_op(
+                        anytype,
+                        func,
+                        op_name="builtin.module",
+                        deduplicate=True,
+                    )
+                    mod = apply_registered_pass(mod, "one-shot-bufferize")
+                    mod = apply_registered_pass(mod, "convert-linalg-to-loops")
+                    cse(mod)
+                    canonicalize(mod)
+
+                    if dump_kernel == "bufferized":
+                        transform.YieldOp()
+                        return schedule_module
```

Comment:
Do we want this in the committed version?

An alternative to consider is inserting `transform.print %mod` if some verbosity flag has been set.